### PR TITLE
[sfml] Add different libraries as features

### PR DESCRIPTION
**Pull request by simolis3**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] ~~SHA512s are updated for each updated download.~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

Add possibility to build SFML with only needed components (graphics, window, audio and network) when required. Default features include all to preserve default functionality the same.

### CODE REVIEW

File: ports/sfml/portfile.cmake [hunk 1]
```diff
@@ -14,6 +14,15 @@ if(VCPKG_TARGET_IS_LINUX)
     message(STATUS "SFML currently requires the following libraries from the system package manager:\n    libudev\n    libx11\n    libxrandr\n    libxcursor\n    opengl\n\nThese can be installed on Ubuntu systems via apt-get install libx11-dev libxrandr-dev libxcursor-dev libxi-dev libudev-dev libgl1-mesa-dev")
 endif()
 
+vcpkg_check_features(
+    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+        "network" SFML_BUILD_NETWORK
+        "graphics" SFML_BUILD_GRAPHICS
+        "window" SFML_BUILD_WINDOW
+        "audio" SFML_BUILD_AUDIO
```
Comment (dg0yt) [inline ports/sfml/portfile.cmake:23]:
```suggestion
        "network"  SFML_BUILD_NETWORK
        "graphics" SFML_BUILD_GRAPHICS
        "window"   SFML_BUILD_WINDOW
        "audio"    SFML_BUILD_AUDIO
```

File: ports/sfml/portfile.cmake [hunk 2]
```diff
@@ -22,6 +31,8 @@ vcpkg_cmake_configure(
         -DSFML_MISC_INSTALL_PREFIX=share/sfml
         -DSFML_GENERATE_PDB=OFF
         -DSFML_WARNINGS_AS_ERRORS=OFF #Remove in the next version
+        -DSFML_INSTALL_PKGCONFIG_FILES=OFF
```
Comment (dg0yt) [inline ports/sfml/portfile.cmake:34]: This is a step backwards.
```suggestion
```

File: ports/sfml/usage [hunk 3]
```diff
@@ -1,7 +1,7 @@
 The package sfml provides CMake targets:
 
     find_package(SFML COMPONENTS system window graphics CONFIG REQUIRED)
-    target_link_libraries(main PRIVATE sfml-system sfml-network sfml-graphics sfml-window)
+    target_link_libraries(main PRIVATE sfml-system sfml-network sfml-graphics sfml-window sfml-audio)
```
Comment (dg0yt) [inline ports/sfml/usage:4]: Is `sfml-audio` available when it is not in `COMPONENTS`?

File: ports/sfml/vcpkg.json [hunk 4]
```diff
@@ -20,5 +14,45 @@
...
+        "libflac",
+        "libogg",
+        "libvorbis",
+        "openal-soft"
+      ]
+    },
+    "graphics": {
+      "description": "Use sfml-graphics library",
+      "dependencies": [
+        "freetype",
```
Comment (dg0yt) [inline ports/sfml/vcpkg.json:37]:
```suggestion
        {
          "name": "freetype",
          "default-features": false
        },
```

File: ports/sfml/portfile.cmake (hunk 2 above)
Comment (simolis3) [reply ports/sfml/portfile.cmake:34]: Removing this line gives build error on Ubuntu 22.04 when trying to use for example only network feature. Relevant part of build output attached: 
[sfml-build-out.txt](https://github.com/user-attachments/files/17584851/sfml-build-out.txt)
It seems to be looking for all the components *.pc files nevertheless. Any good ideas on how to fix this?

File: ports/sfml/usage (hunk 3 above)
Comment (simolis3) [reply ports/sfml/usage:4]: Good point, fixed

File: ports/sfml/portfile.cmake (hunk 2 above)
Comment (simolis3) [reply ports/sfml/portfile.cmake]: Proposed fix in 3b4924f938ede466a78d1b632452c7859075f61e

**Comment by JonLiu1993** [comment]
Tested all features successfully in the following triplet:
- x86-windows
- x64-windows
- x64-widdows-static

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**